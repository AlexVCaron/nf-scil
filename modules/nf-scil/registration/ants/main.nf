

process REGISTRATION_ANTS {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_1.6.0.sif':
        'scilus/scilus:1.6.0' }"

    input:
    tuple val(meta), path(fixedimage), path(movingimage)

    output:
    tuple val(meta), path("*_warped.nii.gz")                                       , emit: image
    tuple val(meta), path("*_GenericAffine.mat"), path("*_warped.nii.gz")          , emit: transfo_image
    tuple val(meta), path("*_Inversewarped.nii.gz"), path("*_GenericAffine.mat")   , emit: transfo_trk
    path "versions.yml"                                                            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        : \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        : \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
