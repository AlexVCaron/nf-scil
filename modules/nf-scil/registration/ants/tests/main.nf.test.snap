{
    "registration - ants": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__warped.nii.gz:md5,131b15551e4c2135b5d2beaf99b77b53"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0Warp.nii.gz:md5,8386e2afbc21d754877bcd75fc8edefa",
                            "test__output1GenericAffine.mat:md5,95fc005e7f2b4de689f3246b14b652c0"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0InverseAffine.mat:md5,aa2254ecad64dcf4e931f3a6f6420039",
                            "test__output1InverseWarp.nii.gz:md5,6cfc53dc70a840be349f75131c82ec88"
                        ]
                    ]
                ],
                "3": [
                    "versions.yml:md5,e1ac7994c0f0169d932e94c6e97a4894"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__warped.nii.gz:md5,131b15551e4c2135b5d2beaf99b77b53"
                    ]
                ],
                "transfo_image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0Warp.nii.gz:md5,8386e2afbc21d754877bcd75fc8edefa",
                            "test__output1GenericAffine.mat:md5,95fc005e7f2b4de689f3246b14b652c0"
                        ]
                    ]
                ],
                "transfo_trk": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0InverseAffine.mat:md5,aa2254ecad64dcf4e931f3a6f6420039",
                            "test__output1InverseWarp.nii.gz:md5,6cfc53dc70a840be349f75131c82ec88"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e1ac7994c0f0169d932e94c6e97a4894"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T13:34:22.61216"
    },
    "registration - ants - quick": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__warped.nii.gz:md5,432be573ac1dfd41893b97ca57e74101"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0Warp.nii.gz:md5,5698bf737986f42a5d6db22efcf4d56b",
                            "test__output1GenericAffine.mat:md5,866d972479304a7b370c1121cba7e42e"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0InverseAffine.mat:md5,e2f5c0b1bbc31c86ca7370b8b1ce53d2",
                            "test__output1InverseWarp.nii.gz:md5,3ec4d027a1d799a7ef48ca401c943873"
                        ]
                    ]
                ],
                "3": [
                    "versions.yml:md5,e1ac7994c0f0169d932e94c6e97a4894"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__warped.nii.gz:md5,432be573ac1dfd41893b97ca57e74101"
                    ]
                ],
                "transfo_image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0Warp.nii.gz:md5,5698bf737986f42a5d6db22efcf4d56b",
                            "test__output1GenericAffine.mat:md5,866d972479304a7b370c1121cba7e42e"
                        ]
                    ]
                ],
                "transfo_trk": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test__output0InverseAffine.mat:md5,e2f5c0b1bbc31c86ca7370b8b1ce53d2",
                            "test__output1InverseWarp.nii.gz:md5,3ec4d027a1d799a7ef48ca401c943873"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e1ac7994c0f0169d932e94c6e97a4894"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T13:35:00.392787"
    },
    "registration - ants - options": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__warped.nii.gz:md5,58007213474153da113836b0b98de428"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output1GenericAffine.mat:md5,73802271efb0a0519ca40901a1be480b"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output0InverseAffine.mat:md5,eaf72200cee6f4843ff8b802738374b7"
                    ]
                ],
                "3": [
                    "versions.yml:md5,e1ac7994c0f0169d932e94c6e97a4894"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__warped.nii.gz:md5,58007213474153da113836b0b98de428"
                    ]
                ],
                "transfo_image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output1GenericAffine.mat:md5,73802271efb0a0519ca40901a1be480b"
                    ]
                ],
                "transfo_trk": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output0InverseAffine.mat:md5,eaf72200cee6f4843ff8b802738374b7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e1ac7994c0f0169d932e94c6e97a4894"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T13:35:24.798892"
    }
}